### PR TITLE
Add go1.19 and go1.20

### PR DESCRIPTION
**Pull request by austinvazquez**
*Issue #, if available:*
#416 

*Description of changes:*
Manually added Go 1.19 and Go 1.20 to the CI matrix and bumped the minimum Go version to Go 1.19. Ran `go mod tidy` and `go mod vendor` using a Go 1.20 compiler. For testing validation the CI still passes.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
   build:
     strategy:
       matrix:
-        go: ['1.15', '1.16', '1.17', '1.18']
+        go: ['1.18', '1.19']
```
Comment (swagatbora90) [inline .github/workflows/build.yaml:13]: Should we add [1.20](https://go.dev/doc/go1.20) along with 1.19?

File: ecr-login/vendor/modules.txt
Comment (sbuckfelder) [inline ecr-login/vendor/modules.txt:1]: Were these changes generated?

**Comment by sbuckfelder** [comment]
I see the majority of this PR is deleted files.  Could you explain in the description which changes were manually done, and which were generated.  In addition, was any testing done to give you confidence this worked?

**Review by ginglis13 — APPROVED** [review]

**Comment by austinvazquez** [comment]
> I see the majority of this PR is deleted files. Could you explain in the description which changes were manually done, and which were generated. In addition, was any testing done to give you confidence this worked?

Updated the description with the details requested.

**Comment by ginglis13** [comment]
This PR is implicitly upgrading the minimum Go version required, right? If so, could you update the README and commit msg to reflect that?

**Comment by austinvazquez** [comment]
> This PR is upgrading the minimum Go version supported and required, right? If so, could you update the README to reflect that? I see mention of using Go 1.16. Maybe could put a note of it in the Prerequisites section

Updated. ptal.

**Review by ginglis13 — APPROVED** [review]
LGTM 👍

**Comment by austinvazquez** [comment]
Tried to close/re-open to get missing checks would go away. Looks like I am not that lucky. 😅

File: ecr-login/go.mod [hunk 2]
```diff
@@ -13,4 +13,20 @@ require (
 	github.com/stretchr/testify v1.8.2
 )
```
Comment (swagatbora90) [inline ecr-login/go.mod:15]: looks like it is missing `go 1.19` to set the min supported go version?

Comment (swagatbora90) [reply ecr-login/go.mod:15]: ahh, it is in L32. I usually expect the go directive to be present after `module`. can you move that here for clarity?

**Review by swagatbora90 — APPROVED** [review]
Minor comment about placement of go directive in go.mod. Otherwise LGTM.

**Review by ginglis13 — APPROVED** [review]

**Pull request merged**